lib/pengine/tests/native/native_find_rsc_test.c

```diff
@@ -0,0 +1,683 @@
+    assert_ptr_equal(promotable_1, native_find_rsc(promotable_1, "promotable-rsc", cluster01, pe_find_any|pe_find_current));
+    assert_ptr_equal(promotable_1, native_find_rsc(promotable_1, "promotable-rsc", cluster01, pe_find_anon|pe_find_current));
+    assert_null(native_find_rsc(promotable_1, "promotable-rsc", cluster02, pe_find_any|pe_find_current));
+    assert_null(native_find_rsc(promotable_1, "promotable-rsc", cluster02, pe_find_anon|pe_find_current));
+
+    /* Fails because incorrect flags were given along with primitive name. */
+    assert_null(native_find_rsc(promotable_0, "promotable-rsc", NULL, pe_find_current));
+    assert_null(native_find_rsc(promotable_1, "promotable-rsc", NULL, pe_find_current));
+
+    /* And then we check failure possibilities again, except passing promotable_clone
+     * instead of promotable_X as the first argument to native_find_rsc.
+     */
+
+    /* Fails because pe_find_current is required if a node is given. */
+    assert_null(native_find_rsc(promotable_clone, "promotable-rsc:0", cluster02, 0));
+    assert_null(native_find_rsc(promotable_clone, "promotable-rsc:1", cluster01, 0));
+
+    /* Check that the resource is running on the node we expect. */
+    assert_ptr_equal(promotable_0, native_find_rsc(promotable_clone, "promotable-rsc:0", cluster02, pe_find_current));
+    assert_ptr_equal(promotable_0, native_find_rsc(promotable_clone, "promotable-rsc", cluster02, pe_find_any|pe_find_current));
+    assert_ptr_equal(promotable_0, native_find_rsc(promotable_clone, "promotable-rsc", cluster02, pe_find_anon|pe_find_current));
+    assert_ptr_equal(promotable_1, native_find_rsc(promotable_clone, "promotable-rsc:1", cluster01, pe_find_current));
+    assert_ptr_equal(promotable_1, native_find_rsc(promotable_clone, "promotable-rsc", cluster01, pe_find_any|pe_find_current));
+    assert_ptr_equal(promotable_1, native_find_rsc(promotable_clone, "promotable-rsc", cluster01, pe_find_anon|pe_find_current));
+}
+
+static void
+renamed_rsc(void **state) {
+    pe_resource_t *promotable_0 = NULL;
+    pe_resource_t *promotable_1 = NULL;
+
+    /* Find the "promotable-rsc:0" and "promotable-rsc:1" resources, members of "promotable-clone". */
+    for (GList *iter = promotable_clone->children; iter != NULL; iter = iter->next) {
+        pe_resource_t *rsc = (pe_resource_t *) iter->data;
+
+        if (strcmp(rsc->id, "promotable-rsc:0") == 0) {
+            promotable_0 = rsc;
+        } else if (strcmp(rsc->id, "promotable-rsc:1") == 0) {
+            promotable_1 = rsc;
+        }
+    }
+
+    assert_non_null(promotable_0);
+    assert_non_null(promotable_1);
+
+    /* Passes because pe_find_renamed means the base name matches clone_name. */
+    assert_ptr_equal(promotable_0, native_find_rsc(promotable_0, "promotable-rsc", NULL, pe_find_renamed));
+    assert_ptr_equal(promotable_1, native_find_rsc(promotable_1, "promotable-rsc", NULL, pe_find_renamed));
```

Comment:
I wonder if pe_find_renamed is effectively any different from pe_find_clone. We definitely have too many find options, and I'd love to ditch clone_name someday if possible. I'm not sure whether clone_name can be different from ID(rsc->xml); possibly in a corner case like the unique-to-anonymous conversion.

BTW I just noticed that pe_find_clone incorrectly uses pcmk__str_casei, it should be pcmk__str_none -- maybe do that as a commit before the unit test.

Reply:
Fixed the case comparison.  It didn't change the test output.